content/en/service_management/app_builder/queries.md

```diff
@@ -0,0 +1,149 @@
+
+Queries rely on [Connections][5] for authentication. App Builder shares connections with [Workflow Automation][6].
+
+## Run settings
+
+**Run Settings** determine when a query is executed. There are two options:
+
+- **Auto**: The query runs when the app loads and whenever any query arguments change.
+- **Manual**: The query runs when another portion of the app triggers it. For example, use a manual trigger if you want a query to execute only when a user clicks a UI button component. For more information on event triggers, see [Events][11].
+
+## Debounce
+
+Configuring debounce ensures that your query is only triggered once per user input. By default, debounce is set to `0` milliseconds (ms). To prevent a query from being called too frequently, increase the debounce. Configure debounce in the **Advanced** section of a query.
+
+## Conditional queries
+
+You can set a condition that must be met before a query can run. To set the condition for a query, enter an expression in the **Condition** field in the **Advanced** section of the query. This condition must evaluate to true before the query can run. For example, if you want a given query to run only if a UI component named `select0` exists and is not empty, use the following expression:
+
+{{< code-block lang="js" >}}${select0.value && select0.value.length > 0}{{< /code-block >}}
+
+## Post-query transformation
+
+Perform a post-query transformation to simplify or transform the output of a query. Add a post-query transformation in the **Advanced** section of a query.
+
+For example, the Slack _List Channels_ action returns an array of dictionaries containing the ID and name for each channel. To discard the IDs and return only an array of names, add the following query transformation:
+
+{{< code-block lang="js" collapsible="false" >}}
+// Use `outputs` to reference the query's unformatted output.
+// TODO: Apply transformations to the raw query output
+arr = []
+object = outputs.channels
+for (var item in object) {
+    arr.push(object[item].name);
+}
+
+return arr
+{{< /code-block >}}
+
+## Post-query hooks
+
+Similar to UI component events, you can configure a reaction to trigger after a query executes. A **post-query hook** can set a UI component state, open or close a modal, trigger another query, or even run custom JavaScript. For example, the [ECS Task Manager][7] blueprint's `scaleService` query uses a post-query hook to rerun the `describeService` query after it executes.
+
+## Error notifications
+
+To display a toast (a brief notification message) to the user when the system returns an error, toggle **Show Toast on Errors** in the **Advanced** section of a query.
+
+## Confirmation prompts
+
+To prompt a user for confirmation before the query runs, toggle the **Requires Confirmation** option in the **Advanced** section of a query.
```

Comment:
Just a note that none of this content is new, it's the content that moved over from build.md